CRM/Core/BAO/CustomField.php

```diff
@@ -1073,6 +1073,19 @@ private static function formatDisplayValue($value, $field, $entityId = NULL) {
         }
         else {
           $display = CRM_Utils_Array::value($value, $field['options'], '');
+          // For float type (see Number and Money) $value would be decimal like
+          // 1.00 (because it is stored in db as decimal), while options array
+          // key would be integer like 1. In this case expression on line above
+          // would return empty string (default value), despite the fact that
+          // matching option exists in the array.
+          // In such cases we could just get intval($value) and fetch matching
+          // option again, but this would not work if key is float like 5.6.
+          // So we need to truncate trailing zeros to make it work as expected.
+          if ($display === '' && strpos($value, '.') !== FALSE) {
+            // Truncate trailing zeros, e.g: 5.00 -> 5, 5.60 -> 5.6.
+            $value = rtrim($value, '.0');
+            $display = CRM_Utils_Array::value($value, $field['options'], '');
+          }
```

Comment:
@ivan-compucorp How about:
```
        else {
          // For float...
          if ($display === '' && strpos($value, '.') !== FALSE) {
            // Truncate trailing zeros, e.g: 5.00 -> 5, 5.60 -> 5.6.
            $value = rtrim($value, '.0');
          }
          $display = $field['options'][$value] ?? '';
```

Reply:
Looks fine for me. Anything else to update?

Reply:
Looks fine for me. Actually I would prefer to do it this way:
```
$display = isset($field['options'][$value]) ? $field['options'][$value] : '';
```
Anything else to update?

Reply:
@ivan-compucorp you are describing a coalesce operation: `isset($foo) ? $foo : 'bar';` is a coalesce, and it is better written with the coalesce operator: `$foo ?? 'bar';`

So the line would be

    $display = $field['options'][$value] ?? '';

Reply:
@colemanw see comments elsewhere on this - I'm not sure this is the right fix - I think it's a db fix required

Reply:
@colemanw yes I agree that in general we should use `??` approach as it's shorter, at the same time if `$field['options'][$value]` does not exist - php would generate a warning. It would break nothing of course, but to avoid anwanted warnings in log I would prefer to use `isset()` or `!empty()` first to check if value exists. That's why this code looks more preferable:
```
$display = isset($field['options'][$value]) ? $field['options'][$value] : '';
```

Reply:
@ivan-compucorp - coalesce does not generate php warnings. See https://wiki.php.net/rfc/isset_ternary

>The coalesce, or ??, operator is added, which returns the result of its first operand if it exists and is not NULL, or else its second operand. This means the $_GET['mykey'] ?? "" is completely safe and will not raise an E_NOTICE.